src/components/surveys/survey-design/question-edit/rhs-subcontrols/Select.tsx

```diff
@@ -172,30 +217,49 @@ const Select: React.FunctionComponent<{
 
   const handleDataTypeChange = (value: typeof step.baseType, shouldForceChange?: boolean) => {
     if (value !== step.baseType) {
-      if (value === 'integer' && !shouldForceChange && step.other) {
+      const isSwitchToInteger = value === 'integer'
+
+      if (isSwitchToInteger && !shouldForceChange && step.other) {
         setIsTypeConversionWarning(true)
         return
       }
       const updatedStep = {...step}
       updatedStep.baseType = value as typeof step.baseType
-      let choices = [...step.choices]
-      //switch from string to integer -- number them
-      if (value === 'integer') {
-        for (const [i, v] of choices.entries()) {
-          v.value = generateValue(v, i)
-        }
-        //if changing to integer -- remove 'other'
+
+      if (isSwitchToInteger) {
+        // Other field is not valid for an integer
         delete updatedStep.other
-      } else {
-        for (const [_i, v] of choices.entries()) {
-          v.value = generateValue(v)
-        }
       }
 
-      onChange({...updatedStep, choices})
+      // generate the appropriate mapping for switch
+      const changes: ChangeValue[] = step.choices.map((choice, ii) => ({
+        choice: choice, 
+        index: ii,
+        newValue:  generateValue(choice, isSwitchToInteger ? ii : undefined, value)
+      }))
+      
+      // apply the changes to both choices and survey rules
+      const choices = getNewChoices(changes)
+      const surveyRules = getNewRules(changes)
+
+      onChange({...updatedStep, choices, surveyRules})
```

Comment:
Could `updateChoicesAndRules` be updated to take `step` as a parameter, so that function could be called here?